packages/yarnpkg-cli/sources/tools/BaseCommand.ts

```diff
@@ -1,8 +1,16 @@
 import {CommandContext}  from '@yarnpkg/core';
+import {npath, ppath}    from '@yarnpkg/fslib';
 import {Command, Option} from 'clipanion';
 
 export abstract class BaseCommand extends Command<CommandContext> {
   cwd = Option.String(`--cwd`, {hidden: true});
 
   abstract execute(): Promise<number | void>;
+
+  validateAndExecute() {
+    if (typeof this.cwd !== `undefined`)
+      this.context.cwd = ppath.resolve(this.context.cwd, npath.toPortablePath(this.cwd));
```

Comment:
That seems something between a bug and a breaking change: it won't read the content from the `.yarnrc.yml` in the target folder anymore, same with everything else we do before actually running the command (like the `Configuration.find` calls).

Reply:
That said, perhaps we can move all this logic inside `validateAndExecute`, rather than keep it inside `main` 🤔

Reply:
🤔 You're right, but that was always the case for the other commands that modify the cwd (e.g. `yarn ./foo [...]`, `yarn workspace`, `yarn workspaces foreach`). I'd agree it's a bug, but at least this PR makes it consistent across all commands 😄 

The things affected by this are `yarnPath`, `enableTelemetry` and plugins (which will be ignored in the target `--cwd`).

`yarnPath` and telemetry logic could be moved to `BaseCommand`, but the CLI needs to know which plugins to load before it can run.

In this case, I'd rather have both `yarn --cwd foo` and `yarn ./foo` be broken but do the same thing. It would be weirdly inconsistent for one of them to use the right path and load the right plugins while the other doesn't.